python/artm/model.py

```diff
@@ -0,0 +1,723 @@
+
+    def remove_dictionary(self, dictionary_name=None):
+        """ArtmModel.remove_dictionary() --- remove the loaded BigARTM dictionary
+        from the lib
+
+        Args:
+          dictionary_name (str): the name of the dictionary in th lib, default=None
+        """
+        if dictionary_name is not None:
+            self.master.lib_.ArtmDisposeDictionary(self.master.id_, dictionary_name)
+        else:
+            raise IOError('ArtmModel.remove_dictionary(): dictionary_name is None')
+
+    def fit_offline(self, collection_name=None, batches=None, data_path='',
+                    num_collection_passes=1, decay_weight=0.0, apply_weight=1.0,
+                    reset_theta_scores=False, data_format='batches', batch_size=1000):
+        """ArtmModel.fit_offline() --- proceed the learning of
+        topic model in off-line mode
+
+        Args:
+          collection_name (str): the name of text collection (required if
+          data_format == 'bow_uci'), default=None
+          batches (list of str): list of file names of batches to be processed.
+          If not None, than data_format should be 'batches'. Format --- '*.batch',
+          default=None
+          data_path (str):
+          1) if data_format == 'batches' => folder containing batches and dictionary;
+          2) if data_format == 'bow_uci' => folder containing
+            docword.collection_name.txt and vocab.collection_name.txt files;
+          3) if data_format == 'vowpal_wabbit' => file in Vowpal Wabbit format;
+          4) if data_format == 'plain_text' => file with text;
+          default=''
+          num_collection_passes (int): number of iterations over whole given
+          collection, default=1
+          decay_weight (int): coefficient for applying old n_wt counters,
+          default=0.0 (apply_weight + decay_weight = 1.0)
+          apply_weight (int): coefficient for applying new n_wt counters,
+          default=1.0 (apply_weight + decay_weight = 1.0)
+          reset_theta_scores (bool): reset accumulated Theta scores
+          before learning, default=False
+          data_format (str): the type of input data;
+          1) 'batches' --- the data in format of BigARTM;
+          2) 'bow_uci' --- Bag-Of-Words in UCI format;
+          3) 'vowpal_wabbit' --- Vowpal Wabbit format;
+          4) 'plain_text' --- source text;
+          default='batches'
+
+          Next argument has sense only if data_format is not 'batches'
+          (e.g. parsing is necessary).
+            batch_size (int): number of documents to be stored ineach batch,
+            default=1000
+
+        Note:
+          ArtmModel.initialize() should be proceed before first call
+          ArtmModel.fit_offline(), or it will be initialized by dictionary
+          during first call.
+        """
+        if collection_name is None and data_format == 'bow_uci':
+            raise IOError('ArtmModel.fit_offline(): No collection name was given')
```

Comment:
I fully agree on this, but I would like to comment on some differences between C++ vs Python here. 

For native C++ I prefer to add the method name into the exception because call stack is typically not available (and its nearly impossible to get the call stack in a cross-platform fashion). The only case when native call is easily available are crashes :)
Also, note that ``c_interface`` has a translation layer that maps C++ exceptions into an integer error code, and sets last error message. Then this error code is converted into a python exception by the wrapper. No native call stack info is passed from C++ to Python at this point, so all info should be in the exception message itself.